aws/logs_monitoring/lambda_function.py

```diff
@@ -1034,44 +1019,95 @@ def is_cloudtrail(key):
     return bool(match)
 
 
-def parse_event_source(event, key):
-    """Parse out the source that will be assigned to the log in Datadog
-
-    TODO Refactor to do a more thorough parsing that will guarantee correct source identification
+def find_cloudwatch_source(log_group):
+    # e.g. /aws/rds/instance/my-mariadb/error
+    if log_group.startswith("/aws/rds"):
+        for engine in ["mariadb", "mysql"]:
+            if engine in log_group:
+                return engine
+        return "rds"
 
-    Args:
-        event (dict): The AWS-formatted log event that the forwarder was triggered with
-        key (string): The S3 object key if the event is from S3 or the CW Log Group if the event is from CW Logs
-    """
-    lowercase_key = str(key).lower()
+    # e.g. Api-Gateway-Execution-Logs_xxxxxx/dev
+    if log_group.startswith("api-gateway"):
+        return "apigateway"
 
-    if "elasticloadbalancing" in lowercase_key:
+    # e.g. dms-tasks-test-instance
+    if log_group.startswith("dms-tasks"):
+        return "dms"
+
+    # e.g. sns/us-east-1/123456779121/SnsTopicX
+    if log_group.startswith("sns/"):
+        return "sns"
+
+    for source in [
+        "/aws/lambda",  # e.g. /aws/lambda/helloDatadog
+        "/aws/codebuild",  # e.g. /aws/codebuild/my-project
+        "/aws/kinesis",  # e.g. /aws/kinesisfirehose/dev
+        "/aws/docdb",  # e.g. /aws/docdb/yourClusterName/profile
```

Comment:
Would it be clearer if I were to break this into two for-loops? One using `startswith` for the `/aws/<source>` and one that doesn't to be more strict? Like so:

```py
# log group must be prefixed
for source in ["/aws/lambda", "/aws/codebuild", "/aws/kinesis", "/aws/docdb",]:
    if log_group.startswith(source):
        return source.replace("/aws/", "")

# log group must contain
for source in ["route53", "vpc", "fargate", "cloudtrail",]:
    if source in log_group:
        return source
```